week1/homework/src/server.js

```diff
@@ -2,16 +2,20 @@
 
 const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
- */
+const getResponse = (responseStatus, response, responseObj) => {
+  response.writeHead(responseStatus, { 'Content-Type': 'application/json' });
+  response.end(JSON.stringify(responseObj));
+};
+
 function createServer(port) {
   let state = 10;
-
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    if (request.url === '/state') getResponse(200, response, { state: state });
+    else if (request.url === '/add') getResponse(200, response, { state: ++state });
+    else if (request.url === '/subtract') getResponse(200, response, { state: --state });
+    else if (request.url === '/reset') getResponse(200, response, { state: (state = 10) });
```

Comment:
In a case of using a single variable with many expected values, we could use a `switch` statement instead.

It will be a lot more legible for readers and will tell what it may contain of case uses.
And `(state = 10)` will be on it's own line as it's reassigned either ways, but it will be telling it's been reassigned :)

Reply:
Hi Jawhar, thank you so much for your review. I think, the reason why I prefer not to use switch-case is  while we were in the first JavaScript lessons our teacher imposed us 'switch-case is not preferred when you write codes in the work.' :)) Actually I thought to use ternary instead of if else :)

Reply:
Yes that's true, but it's always if you have a variable definition and it's value on of a set of values, I will go for the switch statement.

so you don't have to write always:
```javascirpt
if (variable === '/state') {
} else if (variable === '/other-state') {
} else {
}
```

in a switch it will be as the following which make it a lot easier to read:
```javascirpt
switch (variable) {
  case '/state':
  case '/other-state':
  default :
}
```